### ISSUE TITLE
Dropdown Accessibility

### ISSUE THREAD

**Issue by d-m-s** [vmware/clarity#3521]
## Summary
Some words upfront - I think the efforts regarding accessibility and keyboard-navigation-support you're doing are really great! Especially as these are always hard to sell to stakeholders.

**Related**
_Added keyboard Interaction for a dropdown menu component (#3283)
Dropdown components now support keyboard interactions as part of these accessibility improvements. The documentation has been updated with a new accessibility section containing the details._

The latest change for the dropdown leaves some problems though
![dropdown-a11y](https://user-images.githubusercontent.com/311619/60177904-61ddac80-981a-11e9-8245-ca95794eda29.png)

The current solution leaves the user with two different focus states, 
- one for mouse, 
- one for keyboard navigation. 

This is not only visually somehow odd, but also puts additional cognitive load to the user in term of ... "which is the one i'm currently selecting"

### Suggestion
- A good solution would be a highlighting of the first entry (either when opened with the keyboard, or mouse click)
- but as soon as the user, navigates by mouse over an item of the dropdown - just the one beneath the mouse pointer stays highlighted.
- So in the end always one (and only one) item can own the hover/active state.

**Comment by chlane** [comment]
There should not be device dependent i.e,. mouse or keyboard selected states.

Regardless of whether you are using a mouse or keyboard or mixing the usage of both ...

If It is a single select dropdown it should behave like a <select> element where only one menu option is visually highlighted as you cycle through the options with the arrow keys. Additionally, you can only select one option with mouse and it the visual treatment is the same regardless of which devices is used to select the option.

Try it at:
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select

If It is a multiple select dropdown it should behave like a <select> element where you can select multiple options by combining the Shift key and the the arrow keys. To select multiple option with the mouse, you have to use the Ctrl (windows)  or Command (Mac) buttons to select multiple options.

Try it at:
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select_multiple

**Comment by chlane** [comment]
This is not an A11y issue. Nothing in WCAG 2.1 says the hover and focus styling cannot be the same.

**Comment by gnomeontherun** [comment]
Closing this as a duplicate of other ticket #3602.

**Comment by d-m-s** [comment]
@gnomeontherun i fail to see why this should be a duplicate of #3602 ...
probably the suggestion is less an a11y issue, than an usability improvement in general.

what is the purpose from a user's perspective in having two focusable states? 
one for mouse and one for keyboard???
is there any user that opens a dropdown by keyboard and then navigates by mouse or vice-versa?

**Comment by clane** [comment]
@d-m-s Only one element can have focus i.e., be identified by document.activeElement at any given time. Hover is a separate state.  There are users with disabilities that prefer to use the mouse but cannot use it exclusively. Fatigue and dexterity issues are contributing factors. These users will need to be able to distinguish the focus and hover states from one another. If they cannot, our software will not be in compliance with https://www.w3.org/TR/WCAG21/#focus-visible.

**Comment by gnomeontherun** [comment]
No it really wasn't a duplicate but we did change so that the hover and focus states are different. Now the focus state is the browser default, but the hover state is the background highlight. See https://github.com/vmware/clarity/pull/3661 which should address this.

**Comment by d-m-s** [comment]
alright, thx for the quick-reply and background-info @clane & @gnomeontherun !
having different visual representations is a good solution i think!
